data/crac/crac-api/src/main/java/com/farao_community/farao/data/crac_api/cnec/BranchCnec.java

```diff
@@ -9,63 +9,36 @@
 
 import com.farao_community.farao.commons.PhysicalParameter;
 import com.farao_community.farao.commons.Unit;
-import com.farao_community.farao.data.crac_api.Side;
 import com.farao_community.farao.data.crac_api.threshold.BranchThreshold;
 
 import java.util.Optional;
 import java.util.Set;
 
 /**
- * Specific type of {@link Cnec} that is defined on a branch of the network. It presents the singularity of having
- * two sides that could have different voltage levels and independent {@link BranchThreshold}. These thresholds
- * will limit the physical parameter defined for this {@code BranchCnec}.
+ * Specific type of {@link Cnec} that is defined on a branch of the network.
+ *
+ * It presents the singularity of having two sides that could have different voltage levels
+ * and independent {@link BranchThreshold}. These thresholds will limit the {@link PhysicalParameter}
+ * defined for this {@code BranchCnec}.
  *
  * @author Joris Mancini {@literal <joris.mancini at rte-france.com>}
  */
-public interface BranchCnec extends Cnec<BranchCnec> {
+public interface BranchCnec<T extends BranchCnec<T>> extends Cnec<T> {
 
     /**
-     * Getter of the {@link BranchThreshold} that limit the physical parameter of this {@code BranchCnec}.
-     *
-     * @return The set of {@link BranchThreshold} defined on the {@code BranchCnec}.
+     * Getter of the {@link BranchThreshold} below which the {@link PhysicalParameter} of this {@code BranchCnec}
+     * should ideally remain.
```

Comment:
description is inaccurate: this methods can return several thresholds and a threshold can specify a minimum value above which the Physical Parameter should remain.

replace by
"
Getter of the {@link BranchThreshold}s that the {@link PhysicalParameter} of this {@code BranchCnec} should ideally meet."

Reply:
corrected